qiskit/primitives/base/base_estimator.py

```diff
@@ -90,16 +92,18 @@
 from qiskit.circuit.parametertable import ParameterView
 from qiskit.opflow import PauliSumOp
 from qiskit.providers import JobV1 as Job
-from qiskit.providers import Options
 from qiskit.quantum_info.operators import SparsePauliOp
 from qiskit.quantum_info.operators.base_operator import BaseOperator
 from qiskit.utils.deprecation import deprecate_arguments, deprecate_function
 
+from .base_primitive import BasePrimitive
 from .estimator_result import EstimatorResult
-from .utils import _circuit_key, _observable_key, init_observable
+from ..utils import _circuit_key, _observable_key, init_observable
 
+Sequence.register(np.ndarray)
```

Comment:
You can't do this - Numpy would do it if it were valid. `ndarray` doesn't fulfil all the methods of `Sequence` (e.g. `index`) so it isn't one. I haven't read the PR fully, but on a quick scan this seems to be used as a type check only, before you normalise to `tuple`. If you're normalising it to a tuple, you don't need something to be `Sequence`, but generally too, this sort of super-precise type checking isn't necessary.  You've already got the type hinting and documentation, and the `tuple` calls will also give a sensible error message if something fails. Overdoing this just causes slowdown for every user who _is_ doing the right thing. As you're finding here, it also means you originally cut out a bunch of valid inputs, and who's to say it's only `ndarray`?

In general, we also should not be affecting global state like this for any objects we don't own. It's not our decision to make for users.

Reply:
Thanks so much @jakelishman , I was concerned about `Sequence.register` but under the impression that we had to do it. Glad that you stepped in, it is already fixed 🙂 

Regarding parsing the inputs, we had internal conversations about it and your same concerns were raised. I will fill you in on the details on why we decided to go for this separately, and see what your opinions are.

Reply:
It's ok, my stylistic opinions aren't generally relevant here, since I'm not a principal maintainer of these sections. I only commented at all because there was a technical issue I'd seen.